### PR TITLE
Add additional regex pattern for HostDown error code

### PR DESCRIPTION
**Pull request by Daniel-McCarthy**
This pull request is made to assist Desktop's [Issue #926](https://github.com/desktop/desktop/issues/926) by adding a regex pattern to detect a cloning failure error. Currently the error is not being parsed due to not having a pattern that matches to it. The error in question occurs when attempting to clone a repository with no available internet connection.

The exact error to be detected is: 
`"Cloning into '/home/hellokitty/Documents/GitHub/Chixel'... fatal: unable to access 'https://github.com/Daniel-McCarthy/Chixel.git/': Could not resolve host: github.com"`
___
In [a comment](https://github.com/desktop/desktop/issues/926#issuecomment-468799999) it was suggested that this regular expression pattern should be added to target the same error type `HostDown` that is being used by a currently existing regex pattern. I have added a test to go along with checking the new regular expression pattern is parsing properly. However, I noticed there was not already one in place for the original `HostDown` error regex. I could add a test to go along with the original pattern as well in this pull request.

### CODE REVIEW

File: test/fast/git-process-test.ts [hunk 1]
```diff
@@ -423,5 +423,12 @@ mark them as resolved using git add`
       const error = GitProcess.parseError(stderr)
       expect(error).toBe(GitError.UnresolvedConflicts)
     })
+
+    it('can parse the could not resolve host error', () => {
+      const stderr = `"Cloning into '/cloneablepath/'... fatal: unable to access 'https://github.com/Daniel-McCarthy/dugite.git/': Could not resolve host: github.com"`
```
Comment (shiftkey) [inline test/fast/git-process-test.ts]: Are you able to include the newline characters in this `stderr` output so we can confirm the regex is working as expected?

Comment (Daniel-McCarthy) [reply test/fast/git-process-test.ts]: Ah yes, good catch! I have updated both as appropriate.

**Comment by shiftkey** [comment]
>  However, I noticed there was not already one in place for the original `HostDown` error regex. I could add a test to go along with the original pattern as well in this pull request.

Because this is an additive regex I'm fine to leave it as-is.

File: lib/errors.ts [hunk 2]
```diff
@@ -61,6 +61,8 @@ export const GitErrorRegexes = {
   'The requested URL returned error: 403': GitError.HTTPSAuthenticationFailed,
   'fatal: The remote end hung up unexpectedly': GitError.RemoteDisconnection,
   "fatal: unable to access '(.+)': Failed to connect to (.+): Host is down": GitError.HostDown,
+  "Cloning into '(.+)'...\nfatal: unable to access '(.+)': Could not resolve host: (.+)\n":
```
Comment (shiftkey) [inline lib/errors.ts]: For the most part these regexes can be relaxed to ignore control characters at the ends of strings, unless we start exploring returning values included in the groups found in each regex match:

```suggestion
  "Cloning into '(.+)'...\nfatal: unable to access '(.+)': Could not resolve host: (.+)":
```

**Comment by shiftkey** [comment]
`dugite@1.86.0` has been published to NPM with this change - thanks for the contribution @Daniel-McCarthy, and feel free to submit any extra PRs to help with your work on Desktop!

**Comment by Daniel-McCarthy** [comment]
@shiftkey That's awesome, thank you! I will definitely be keeping that in mind.

**Pull request merged**